normandy/recipes/api/serializers.py

```diff
@@ -52,6 +71,21 @@ def update(self, instance, validated_data):
                 'action': Action.objects.get(name=validated_data['action'])
             })
 
+        if 'channels' in validated_data:
```

Comment:
We need the custom `update` method to call `Recipe.update` which is responsible for creating the revision.